src/chains/ethereum/ethereum/tests/api/evm/evm.test.ts

```diff
@@ -79,90 +80,153 @@ describe("api", () => {
-        let customBatchCalled = false;
-        db._batch = (...args) => {
-          setTimeout(() => {
-            customBatchCalled = true;
-            Reflect.apply(memdown.prototype._batch, db, args);
-          }, 20);
-        };
-
-        const options = { database: { db } };
-        const provider = await getProvider(options);
-        await provider.request({ method: "evm_mine", params: [{}] });
-
-        const block = await provider.request({
-          method: "eth_getBlockByNumber",
-          params: [`0x1`]
+      const providerOptions: EthereumProviderOptions[] = [
+        { miner: { instamine: "eager" } },
+        { miner: { instamine: "strict" } }
+      ];
+      providerOptions.forEach(option => {
+        return describe(`in ${option.miner.instamine} instamine mode`, () => {
+          it("should mine `n` blocks on demand", async () => {
+            const provider = await getProvider(option);
+            const initialBlock = parseInt(
+              await provider.send("eth_blockNumber")
+            );
+            await provider.request({
+              method: "evm_mine",
+              params: [{ blocks: 5 }]
+            });
+            const currentBlock = parseInt(
+              await provider.send("eth_blockNumber")
+            );
+            assert.strictEqual(currentBlock, initialBlock + 5);
+          });
+
+          it("should mine `n` blocks if called `n` times without awaiting", async () => {
+            const provider = await getProvider(option);
+            const initialBlock = parseInt(
+              await provider.send("eth_blockNumber")
+            );
+            let proms = [];
+            for (let i = 0; i < 5; i++) {
+              proms.push(
+                provider.request({
+                  method: "evm_mine",
+                  params: []
+                })
+              );
+            }
+
+            await Promise.all(proms);
+            const currentBlock = parseInt(
+              await provider.send("eth_blockNumber")
+            );
+            assert.strictEqual(currentBlock, initialBlock + 5);
+          });
+
+          it("should mine a block on demand", async () => {
+            const provider = await getProvider(option);
+            const initialBlock = parseInt(
+              await provider.send("eth_blockNumber")
+            );
+            await provider.send("evm_mine");
+            const currentBlock = parseInt(
+              await provider.send("eth_blockNumber")
+            );
+            assert.strictEqual(currentBlock, initialBlock + 1);
+          });
+
+          it("should mine a block on demand at the specified timestamp", async () => {
+            const startDate = new Date(2019, 3, 15);
+            const miningTimestamp = Math.floor(
+              new Date(2020, 3, 15).getTime() / 1000
+            );
+            const provider = await getProvider({
+              chain: { time: startDate },
+              ...option
```

Comment:
In some cases I have to merge our "instamine" options with other options needed for the test.